File: src/main/java/com/prography/zone_2_be/global/health/HealthCheckController.java
package com.prography.zone_2_be.global.health;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.prography.zone_2_be.global.response.ApiResponse;

@RestController
@RequestMapping("/api/v1")
public class HealthCheckController {

	@GetMapping("/health")
	public ResponseEntity<ApiResponse<Void>> health() {
		return ApiResponse.success();
	}
}

File: src/main/kotlin/com/prography/zone_2_be/global/health/HealthCheckController.kt
package com.prography.zone_2_be.global.health

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1")
class HealthCheckController {

    @GetMapping("/health")
    fun health(): ResponseEntity<String> =
        ResponseEntity.ok("OK");
}